library/network/src/modules/firewalld_wrapper.rb

```diff
@@ -42,9 +42,25 @@ def write_only
       firewalld.write_only
     end
 
-    publish function: :read, type: "boolean (string)"
-    publish function: :write, type: "boolean (string)"
-    publish function: :write_only, type: "boolean (string)"
+    def add_port(port_range, protocol, interface)
+      zone = firewalld.zones.find { |z| z.interfaces.include?(interface) }
+      return unless zone
+      port = "#{port_range.sub(":", "-")}/#{protocol.downcase}"
```

Comment:
According to 1st parameter name, port range is expected. I'd like to see an argument check to be present. I mean something like ```/[0-9]+:[0-9]+/ =~ port_range```

Reply:
I will document it with examples, just added a quick one to work with http-server. It can be really a por or port_range so it could be 80 or 8080:8090